service/history/mutableStateBuilder.go

```diff
@@ -105,7 +105,9 @@ type (
 		executionInfo    *persistence.WorkflowExecutionInfo // Workflow mutable state info.
 		versionHistories *persistence.VersionHistories
 		replicationState *persistence.ReplicationState
-		hBuilder         *historyBuilder
+		// TODO: persistent this data to db
+		appliedReappliedEvents map[persistence.ReappliedEventRecord]struct{} // record if a event has been reapplied
```

Comment:
for in mem only dedup, plz move this to section under `// in memory only attributes`

Reply:
removed